felix/bpf/polprog/pol_prog_builder_test.go

```diff
@@ -88,15 +90,150 @@ func TestLogActionIgnored(t *testing.T) {
 					Action: "Log",
 				}}},
 			}},
-		}}}, false)
+		}}})
 	Expect(err).NotTo(HaveOccurred())
 
-	pg = NewBuilder(alloc, 1, 2, 3)
+	pg = NewBuilder(alloc, 1, 2, 3, false)
 	noOpInsns, err := pg.Instructions(Rules{
 		Tiers: []Tier{{
 			Name:     "default",
 			Policies: []Policy{},
-		}}}, false)
+		}}})
 	Expect(err).NotTo(HaveOccurred())
 	Expect(noOpInsns).To(Equal(insns))
 }
+
+func TestPolicyDump(t *testing.T) {
+	RegisterTestingT(t)
+	checkLabelsAndComments := func(insns asm.Insns, idx int, expectedString string, matchLabelOrComment string, offset int) {
```

Comment:
I think idx and offset might be overfitting slightly.  Having to adjust them all when we add a new instruction would be a pain?  OK to have one or two tests that verify the exact output but having every test do it makes a lot of places to fix up?

Reply:
I too thought it will be a problem to maintain. Now I have modified the tests so that we don't use idx, offset anymore